### ISSUE TITLE
Restore doesn't work with rails 3.2.11

### ISSUE THREAD

**Issue by mlabarca** [rubysherpas/paranoia#253]
Hi, first of all, I apologize if this issue is unnecessary, I'm still new to rails therefore I could've made a mistake on installation of this gem.  

I couldn't get restore  to work with my rails version, using paranoia ver. 1.3.4 . After going through the code and some past issues such  as #37 it seems update_column still uses default_scope in my rails version. Thanks to the workaround mentioned in #37 so far I have this workaround initializer: 

```
module Paranoia
  # override
  def restore!(opts = {})
    ActiveRecord::Base.transaction do
      run_callbacks(:restore) do
        self.class.unscoped.update_all({deleted_at: nil}, self.class.primary_key => id)
        restore_associated_records if opts[:recursive]
      end
    end
  end
  alias :restore :restore!
end
```

I was wondering if there was a better workaround for this. I also think the readme for the rails 3 should mention this issue referencing the required rails ver. for restore to work as intended. It still is possible to restore a single record scoping the model, i.e. Client.with_deleted.restore(1), but this won't work for its associations using recursive: true.

**Comment by radar** [comment]
Can you please provide steps to reproduce as per the [Contributing Guide](https://github.com/radar/paranoia/blob/rails4/CONTRIBUTING.md#filing-an-issue)

**Comment by mlabarca** [comment]
Hi, what I did was to soft delete a model instance that had a couple of depedant: destroy associations, and each had acts_as_paranoia activated. Soft deletion worked okay, with the parent instance being soft deleted and each chained association. Restoring the instance with something like Model.restore(id) failed and returned a not found error, and looking at the queries it seems the Where part of the Update query is using the altered default scope ( WHERE deleted_at = NULL) thus not updating anything. 

I ran the tests in the gem installation dir and got a bunch of failing tests related to restore: https://gist.github.com/mlabarca/8be7a1f45878be469a14
Also, here's my gemfile: 
https://gist.github.com/mlabarca/6a864dd5abb000638ede
Hope that helps.

**Comment by radar** [comment]
Show me how I can reproduce this issue on a brand new app please.

**Comment by mlabarca** [comment]
On  a brand new app? Since the tests failed, I would say that for  any app with rails 3.2.11 and under and the current rails3  version of the gem, restoring any soft deleted model instance wouldn't work. I think the issue with update_column and the default scope was fixed in rails 3.2.13 though: http://weblog.rubyonrails.org/2013/2/27/Rails-3-2-13-rc1-has-been-released/ .

**Comment by BenMorganIO** [comment]
Closing as Rails 3 isn't supported anymore. Please let me know if this is still an issue in Rails 5
